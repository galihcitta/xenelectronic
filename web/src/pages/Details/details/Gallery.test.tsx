import React from 'react';
import { render, fireEvent, wait, act } from '@testing-library/react';
import { GalleryProps, Gallery } from './Gallery.component';

describe('<Gallery />', () => {
  const props: GalleryProps = {
    imageUrls: ['a', 'b', 'c']
  };

  const setup = (props: GalleryProps) => {
    const utils = render(<Gallery {...props} />);
    const preview = utils.getByTestId('Preview');
    const images = utils.container.querySelectorAll('img');

    return { ...utils, preview, images };
  };

  it('renders correctly', () => {
    const { asFragment } = setup(props);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('preview image', () => {
    it('contains correct style rule and class', () => {
      jest.useFakeTimers();
      const { preview, images } = setup(props);
      expect(preview).toHaveStyle('background-image: url(a)');

      // warning: code that causes React state updates should be wrapped into act(...)
      act(() => {
        fireEvent.click(images[1]);
      });
      expect(preview).toHaveStyle('background-image: url(http://localhost/b)');
      expect(preview).toHaveClass('fade-in');

      act(() => {
        jest.runAllTimers();
      });
      expect(preview).not.toHaveClass('fade-in');
    });
  });
});
