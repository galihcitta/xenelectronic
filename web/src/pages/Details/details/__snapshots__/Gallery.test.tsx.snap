// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa bxSHRL"
  >
    <div
      class="sc-bwzfXH CVjlI"
    >
      <img
        alt=""
        class="sc-htpNat driQCr"
        src="a"
      />
      <img
        alt=""
        class="sc-htpNat driQCr"
        src="b"
      />
      <img
        alt=""
        class="sc-htpNat driQCr"
        src="c"
      />
    </div>
    <div
      class="sc-bxivhb MTqTJ"
      data-testid="Preview"
    />
  </div>
</DocumentFragment>
`;
